sdv/metadata/validation.py

```diff
@@ -23,13 +23,46 @@ def validate_address_sdtypes(columns_to_sdtypes):
     """Validate sdtypes for address column relationship.
 
     Args:
-        - columns_to_sdtypes (dict): Dictionary mapping column names to sdtypes.
+        columns_to_sdtypes (dict):
+            Dictionary mapping column names to sdtypes.
 
     Raises:
-        - ``InvalidMetadataError`` if column sdtypes are invalid for the relationship.
+        ``InvalidMetadataError`` if column sdtypes are invalid for the relationship.
     """
     _check_import_address_transformers()
     try:
         rdt.transformers.address.RandomLocationGenerator._validate_sdtypes(columns_to_sdtypes)
     except TransformerInputError as error:
         raise InvalidMetadataError(str(error))
+
+
+def _check_import_gps_transformers():
+    """Check that the gps transformers can be imported."""
+    error_message = (
+        'You must have SDV Enterprise with the gps add-on to use the gps features'
+    )
+    if not hasattr(rdt.transformers, 'gps'):
+        raise ImportError(error_message)
+
+    has_randomlocationgenerator = hasattr(rdt.transformers.gps, 'RandomLocationGenerator')
+    has_metroaeraanonymizer = hasattr(rdt.transformers.gps, 'MetroAreaAnonymizer')
+    has_gpsnoiser = hasattr(rdt.transformers.gps, 'GPSNoiser')
+    if not has_randomlocationgenerator or not has_metroaeraanonymizer or not has_gpsnoiser:
+        raise ImportError(error_message)
+
+
+def validate_gps_sdtypes(columns_to_sdtypes):
```

Comment:
where is this being called?

Reply:
This is put into a dict here:
https://github.com/sdv-dev/SDV/blob/aca634f82ca5e39625487d3667ff5ff099903711/sdv/metadata/single_table.py#L109

And get called here:
https://github.com/sdv-dev/SDV/blob/aca634f82ca5e39625487d3667ff5ff099903711/sdv/metadata/single_table.py#L708